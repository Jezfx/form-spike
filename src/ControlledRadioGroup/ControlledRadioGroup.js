import React from "react";
import { Controller, useFormContext } from "react-hook-form";
import {
  RadioGroup,
  Radio,
  FormControlLabel,
  FormLabel,
} from "@material-ui/core";

const ControlledRadioGroup = ({ title, defaultValue, group }) => {
  const { control, error } = useFormContext();

  const { key, options } = group;

  return (
    <>
      <FormLabel>{title}</FormLabel>
      <Controller
        render={({
          field: { onChange, value, name },
          fieldState: { error },
        }) => (
          <RadioGroup key={key} name={name} value={value} onChange={onChange}>
            {options?.map((x) => (
              <FormControlLabel
                key={x?.key}
                value={x?.value}
                name={x?.name}
                label={x?.label}
                control={<Radio />}
              />
            ))}
          </RadioGroup>
        )}
        name={key}
        control={control}
      />
    </>
  );
};

export default ControlledRadioGroup;
