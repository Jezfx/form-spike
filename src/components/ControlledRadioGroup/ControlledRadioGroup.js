import React from "react";
import PropTypes from "prop-types";
import { Controller, useFormContext } from "react-hook-form";
import {
  RadioGroup,
  Radio,
  FormControlLabel,
  FormLabel,
  FormControl,
  FormHelperText,
} from "@material-ui/core";

const ControlledRadioGroup = ({ group, name, label }) => {
  const { control, errors } = useFormContext();
  const { options } = group;

  const error = errors[name];

  return (
    <FormControl component="fieldset" required error={!!error}>
      {label && <FormLabel>{label}</FormLabel>}
      <Controller
        name={name}
        control={control}
        render={(
          { onChange, onBlur, value, name, ref },
          { invalid, isTouched, isDirty }
        ) => (
          <RadioGroup
            key={name}
            name={name}
            value={value || ""}
            onChange={onChange}
          >
            {options?.map(({ ...props }) => (
              <FormControlLabel
                key={props.name}
                control={<Radio />}
                {...props}
              />
            ))}
          </RadioGroup>
        )}
      />
      <FormHelperText>{error?.message}</FormHelperText>
    </FormControl>
  );
};

ControlledRadioGroup.propTypes = {
  group: PropTypes.shape({
    key: PropTypes.string,
    options: PropTypes.arrayOf(PropTypes.object),
  }),
  name: PropTypes.string,
};

ControlledRadioGroup.defaultProps = {
  group: {
    key: "",
    options: [],
  },
  name: "",
};

export default ControlledRadioGroup;
