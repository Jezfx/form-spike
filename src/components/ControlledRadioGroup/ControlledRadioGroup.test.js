import React from "react";
import ControlledRadioGroup from "./ControlledRadioGroup";
import { renderWithReactHookForm } from "../../test-utils";

describe("ControlledRadioGroup", () => {
  it("should match snapshot", () => {
    const { asFragment } = renderWithReactHookForm(
      <ControlledRadioGroup options={[]} name="" label="" />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
