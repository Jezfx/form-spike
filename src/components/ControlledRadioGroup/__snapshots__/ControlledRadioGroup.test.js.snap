// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledRadioGroup should match snapshot 1`] = `
<DocumentFragment>
  <fieldset
    class="MuiFormControl-root"
  >
    <div
      class="MuiFormGroup-root"
      role="radiogroup"
    />
    <p
      class="MuiFormHelperText-root Mui-required"
    />
  </fieldset>
</DocumentFragment>
`;
